src/objects/Lotto.js

```diff
@@ -0,0 +1,69 @@
+import { ERROR_MESSAGE, LOTTO_RANK_MAP } from '../lib/constants.js';
+import {
+  getIsAllItemsUnique,
+  getIsArrayLengthMatch,
+  intersection,
+} from '../lib/utils.js';
+
+class Lotto {
+  #numbers;
+
+  constructor(numbers) {
+    this.#numbers = numbers;
+
+    this.#validate();
+    this.#sortByAscending();
+  }
+
+  get numbers() {
+    return this.#numbers;
+  }
+
+  getRank(winningNumbers, bonusNumber) {
+    const winningCount = this.#getWinningNumberCount(winningNumbers);
+    const isBonusMatch = this.#getIsBonusMatch(bonusNumber);
+
+    const rank = Lotto.#calculateRank(winningCount, isBonusMatch);
+    return rank;
+  }
+
+  #validate() {
+    if (!getIsArrayLengthMatch(this.#numbers, 6))
+      throw new Error(ERROR_MESSAGE.NOT_SIX);
+    if (!getIsAllItemsUnique(this.#numbers))
+      throw new Error(ERROR_MESSAGE.NOT_UNIQUE);
+  }
+
+  #sortByAscending() {
+    this.#numbers.sort((a, b) => a - b);
+  }
+
+  #getWinningNumberCount(winningNumbers) {
+    return intersection(this.#numbers, winningNumbers).length;
+  }
+
+  #getIsBonusMatch(bonusNumber) {
+    return intersection(this.#numbers, [bonusNumber]).length > 0;
+  }
+
+  static #calculateRank(winningCount, isBonusMatch) {
+    const rankData = Object.entries(LOTTO_RANK_MAP).find(
+      ([_, info]) =>
+        info.winningCount === winningCount &&
+        info.isBonusMatch === isBonusMatch,
+    );
+
+    if (rankData) return Number(rankData[0]);
```

Comment:
여기서 `Number`로 한번 더 감싸시는 이유가 뭔지 궁금합니다! `rankData[0]`은 이미 `rank`가 숫자로 담겨있지 않나요?

그리구 규성님께서 이번에는 변수에 먼저 담고 그걸 사용함으로써 가독성을 높이신 경우가 많았는데, 여기서도 활용해보시면 어떨까요? 지금은 `rankData[0]`이 어떤걸 의미하는지 알기가 어려운 것 같아요..!
```suggestion
    const [rank, rankInfo] = rankData;
    if (rankData) return Number(rank);
```

Reply:
객체의 key값은 전부 문자열화가 됩니다. 위 rankData는 객체의 key값에 해당합니다. 그래서 rankData의 타입은 String이고, 원하는 Number타입의 순위를 얻기 위해서는 `Number`로 변환해주어야했습니다.

Reply:
호, 그러한 방식은 고려해보지 못했네요. 좋은 거 같아요. 감사합니다 !